pkg/util/commands.go

```diff
@@ -269,14 +269,17 @@ func (c *Command) run() (string, error) {
 // PathWithBinary Sets the $PATH variable. Accepts an optional slice of strings containing paths to add to $PATH
 func PathWithBinary(paths ...string) string {
 	path := os.Getenv("PATH")
-	binDir, _ := JXBinLocation()
-	answer := path + string(os.PathListSeparator) + binDir
 	mvnBinDir, _ := MavenBinaryLocation()
+	answer := ""
 	if mvnBinDir != "" {
 		answer += string(os.PathListSeparator) + mvnBinDir
 	}
 	for _, p := range paths {
-		answer += string(os.PathListSeparator) + p
+		if p != "" {
+			answer += string(os.PathListSeparator) + p
+		}
 	}
+	binDir, _ := JXBinLocation()
```

Comment:
maybe _jxBinDir_ to make it a bit cleaner in the final line where you put things together

Reply:
A test for this might not be bad either?

Reply:
test is incoming, I've made the variables in here clearer